pypeit/images/rawimage.py

```diff
@@ -556,14 +556,19 @@ def spatial_flexure_shift(self, slits, force=False):
                 Force the image to be field flattened, even if the step log
                 (:attr:`steps`) indicates that it already has been.
 
+        Return:
+            float: The calculated flexure correction
+
         """
         step = inspect.stack()[0][3]
         if self.steps[step] and not force:
             # Already field flattened
             msgs.warn('Spatial flexure shift already calculated.')
-            return
+            return self.spat_flexure_shift 
```

Comment:
Can you modify the logic here to be as below, which would be conceptually cleaner.  

if self.steps[step] and not force:
           # Already field flattened
             msgs.warn('Spatial flexure shift already calculated.')
             return self.spat_flexure_shift 
else: 
                    self.spat_flexure_shift = flexure.spat_flexure_shift(self.image, slits)
         self.steps[step] = True
         # Return
         return self.spat_flexure_shift

Reply:
done

Reply:
Seems like either the return statement should be in the else, or not in the if. My suggestion is to just add a tab such that the return is part of the else.